### PR TITLE
MES 4405: Start Cat A MOD 1 Test From Journal Page

### PR DESCRIPTION
**Pull request by faggc**
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/components/test-slot/test-outcome/test-outcome.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { TestStatus } from '../../../modules/tests/test-status/test-status.model
 import { StartE2EPracticeTest } from '../../../pages/fake-journal/fake-journal.actions';
 import { startsWith, isEmpty } from 'lodash';
 import { end2endPracticeSlotId } from '../../../shared/mocks/test-slot-ids.mock';
-import { JOURNAL_FORCE_CHECK_MODAL, CAT_B, CAT_BE, CAT_C } from '../../../pages/page-names.constants';
+import { JOURNAL_FORCE_CHECK_MODAL, CAT_B, CAT_BE, CAT_C, CAT_AMOD1 } from '../../../pages/page-names.constants';
```
Comment (appleabs) [inline src/components/test-slot/test-outcome/test-outcome.ts]: my PR will change the naming of this to CAT_A_MOD1, so this might get broken

Comment (faggc) [reply src/components/test-slot/test-outcome/test-outcome.ts]: Thanks for the heads up

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<ion-header>
+  <practice-mode-banner *ngIf="isPracticeMode"></practice-mode-banner>
```
Comment (szabikr) [inline src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html]: No practice banner needed as not practiceable page currently

File: src/pages/waiting-room/cat-a-mod1/__tests__/waiting-room.cat-a-mod1.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  let screenOrientation: ScreenOrientation;
+  let insomnia: Insomnia;
+  let translate: TranslateService;
+  let navController: NavController;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        WaitingRoomCatAMod1Page,
+        MockComponent(PracticeModeBanner),
```
Comment (szabikr) [inline src/pages/waiting-room/cat-a-mod1/__tests__/waiting-room.cat-a-mod1.page.spec.ts]: Can be removed

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  <ion-navbar>
+    <ion-title>{{ 'waitingRoom.title' | translate }} - {{pageState.candidateUntitledName$ | async}}</ion-title>
+    <ion-buttons end>
+      <end-test-link category="B"></end-test-link>
```
Comment (szabikr) [inline src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html]: Want to see the correct category here

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+import { Observable } from 'rxjs/Observable';
+import { getPreTestDeclarations } from '../../../modules/tests/pre-test-declarations/pre-test-declarations.reducer';
+import * as preTestDeclarationsActions
+  from '../../../modules/tests/pre-test-declarations/pre-test-declarations.actions';
+import {
+  getInsuranceDeclarationStatus,
+  getResidencyDeclarationStatus,
+  getSignatureStatus,
+} from '../../../modules/tests/pre-test-declarations/pre-test-declarations.selector';
+// TODO: Replace reducer when relevant a-mod1 reducer
```
Comment (szabikr) [inline src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts]: Todo has to be labeled with PREP-AMOD1

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+  }
+
+  dispatchCandidateChoseToProceedInEnglish() {
+    this.store$.dispatch(new CandidateChoseToProceedWithTestInEnglish(Language.ENGLISH));
+  }
+
+  onSubmit() {
+    Object.keys(this.formGroup.controls).forEach(controlName => this.formGroup.controls[controlName].markAsDirty());
+    if (this.formGroup.valid) {
+      this.navController.push(CAT_B.COMMUNICATION_PAGE);
```
Comment (szabikr) [inline src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts]: CAT_B needs to change to CAT_AMOD1

**Review by szabikr — APPROVED** [review]
Approving it, but a few comments needs to be addressed before merging

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html (hunk 2 above)
Comment (faggc) [reply src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html]: Done

File: src/pages/waiting-room/cat-a-mod1/__tests__/waiting-room.cat-a-mod1.page.spec.ts (hunk 3 above)
Comment (faggc) [reply src/pages/waiting-room/cat-a-mod1/__tests__/waiting-room.cat-a-mod1.page.spec.ts]: Done

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts (hunk 5 above)
Comment (faggc) [reply src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts]: Done

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts (hunk 6 above)
Comment (faggc) [reply src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.ts]: Done

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<ion-header>
+  <ion-navbar>
+    <ion-title>{{ 'waitingRoom.title' | translate }} - {{pageState.candidateUntitledName$ | async}}</ion-title>
+    <ion-buttons end>
+      <end-test-link category="AMOD1"></end-test-link>
```
Comment (appleabs) [inline src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html]: This should be EUAM1

File: package-lock.json [hunk 8]
```diff
@@ -25770,8 +25713,7 @@
       "version": "1.7.0",
       "resolved": "https://registry.npmjs.org/underscore/-/underscore-1.7.0.tgz",
       "integrity": "sha1-a7rwh3UA02vjTsqlhODbn+8DUgk=",
-      "dev": true,
-      "optional": true
```
Comment (appleabs) [inline package-lock.json]: just curious as to what this is doing does any one know?

File: src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html (hunk 7 above)
Comment (faggc) [reply src/pages/waiting-room/cat-a-mod1/waiting-room.cat-a-mod1.page.html]: Done

**Review by appleabs — APPROVED** [review]
nice!

File: mock/generate-local-journal.ts [hunk 9]
```diff
@@ -556,6 +556,53 @@ const localJournal: ExaminerWorkSchedule = {
...
+          driverNumber: 'LEWIS375220A99HC',
+          gender: 'F',
+          primaryTelephone: '01234 567890',
+          dateOfBirth: '1973-09-06',
+          ethnicityCode: 'A',
+        },
+      },
+      slotDetail: {
+        duration: 57,
+        slotId: 1008,
```
Comment (TomBillingtonUK) [inline mock/generate-local-journal.ts]: This needs to be unique

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**